libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1227,39 +1227,64 @@ pair<smtutil::Expression, smtutil::Expression> SMTEncoder::arithmeticOperation(
 	else
 		intType = TypeProvider::uint256();
 
-	smtutil::Expression valueNoMod(
-		_op == Token::Add ? _left + _right :
-		_op == Token::Sub ? _left - _right :
-		_op == Token::Div ? division(_left, _right, *intType) :
-		_op == Token::Mul ? _left * _right :
-		/*_op == Token::Mod*/ _left % _right
-	);
+	optional<smtutil::Expression> valueUnbounded;
+	if (_op == Token::Add)
+		valueUnbounded = _left + _right;
+	else if (_op == Token::Sub)
+		valueUnbounded = _left - _right;
+	else if (_op == Token::Mul)
+		valueUnbounded = _left * _right;
+	else if (_op == Token::Div)
+		valueUnbounded = division(_left, _right, *intType);
+	else /* _op == Token::Mod */
+		valueUnbounded = _left % _right;
```

Comment:
Wanna try a fancy way to get around the ``optional``?
```suggestion
	auto valueUnbounded = [&]() -> smtutil::Expression {
	  switch(_op)
	  {
	     case Token::Add: return _left + _right;
	     case Token::Sub: return _left - _right;
	     case Token::Mul: return _left * _right;
	     case Token::Div: return division(_left, _right, *intType);
	      // maybe even make this case Token::Mod and assert(false) in default?
	     default: return _left % _right;
	}();
```

Reply:
Nice!